collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -400,9 +402,25 @@ public static void deleteAndReport(File file) {
         }
     }
 
+    public static String getFormFileName(File formXml) {
+        return getFormFileName(formXml.getName());
+    }
+
+    public static String getFormFileName(String formFilePath) {
+        return formFilePath.substring(0, formFilePath.lastIndexOf('.'));
+    }
+
     public static String constructMediaPath(String formFilePath) {
-        String pathNoExtension = formFilePath.substring(0, formFilePath.lastIndexOf('.'));
-        return pathNoExtension + "-media";
+        return getFormFileName(formFilePath) + "-media";
+    }
+
+    public static File getFormMediaDir(File formXml) {
+        final String formFileName = getFormFileName(formXml);
+        return new File(formXml.getParent(), formFileName + "-media");
+    }
+
+    public static File getLastSavedFile(File formXml) {
+        return new File(getFormMediaDir(formXml), LAST_SAVED_FILENAME);
```

Comment:
(minor) Since this has moved here it seems like `LAST_SAVED_FILENAME` should move to `FileUtils` as well?

Reply:
Ha, I see that you already did this!